File: Level1/GCDLCM.java
package Level1;

public class GCDLCM {
    public int[] solution(int n, int m) {

        int max = 0;
        int min = 0;
        // for (int i = 1; i <= n && i <= m; i++) { => better option is
        for (int i = 1; i <= Math.max(m, n); i++) {
            if (n % i == 0 && m % i == 0) {
                max = i;
            }
        }
        min = (n * m) / max;

        int[] answer = { max, min };
        return answer;
    }
}

File: Level1/mathmaticalthink/gcdLCM.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12940?language=python3
def gcdlcm(a, b):
    c, d = max(a, b), min(a, b)

    while d:
        c, d = d, c % d
    answer = [c, int(a*b/c)]

    return answer
